www/js/app.js

```diff
@@ -93,7 +93,9 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     // A parameter to access the URL of any extension that this app was launched from
     params['referrerExtensionURL'] = settingsStore.getItem('referrerExtensionURL');
     // A parameter to set the content injection mode ('jquery' or 'serviceworker') used by this app
-    params['contentInjectionMode'] = settingsStore.getItem('contentInjectionMode') || 'jquery'; // Defaults to jquery for now
+    params['contentInjectionMode'] = settingsStore.getItem('contentInjectionMode') || 
+        // Defaults to jquery in extensions, and serviceworker if accessing as a PWA
+        (/http/i.test(window.location.protocol) && isServiceWorkerAvailable()) ? 'serviceworker' : 'jquery';
```

Comment:
I would suggest to use a more specific regexp, like: 
```suggestion
        (/^https?$/i.test(window.location.protocol) && isServiceWorkerAvailable()) ? 'serviceworker' : 'jquery';
```
It should avoid matching a protocol like `this-is-not-http-protocol` (which does not exist, of course, it's an example)
NB : I did not test

Reply:
OK, will do.